### ISSUE TITLE
clone_on_copy does not suggest dereference

### ISSUE THREAD

**Issue by Diggsey** [rust-lang/rust-clippy#3499]
Regression? of #1139

(v0.0.212)

When initializing a struct, like:

```rust
Foo {
    bar: bar.clone()
}
```

Assuming `bar` is a reference to a `Copy` type, then the `clone_on_copy` lint will fire and suggest removing the `clone()` call, but to do so is not valid, you need to also de-reference `bar`:

```rust
Foo {
    bar: *bar
}
```

**Comment by detrumi** [comment]
I tried to reproduce it, but it correctly suggests `*bar`.
```
struct Foo {
    bar: i32,
}

fn main() {
    let bar = &2;
    Foo {
        bar: bar.clone()
    };
}
```

Either it's fixed already, or I'm not hitting the problem. Can you give a full example, @Diggsey ?

**Comment by Diggsey** [comment]
Hmm, it seems to work correctly in at least some situations, and I don't have the example to hand any more (was fixing a ton of clippy errors at once and didn't want to spend too long on each), will close this for now and re-open if it comes up again.
